plumbing/transport/http/common.go

```diff
@@ -145,6 +145,25 @@ type AuthMethod interface {
 	setAuth(r *http.Request)
 }
 
+// TokenAuthMethod is concrete implementation of common.AuthMethod for HTTP services
+// Allow Bearer Token used in git authentication.
+type TokenAuth struct {
+	token string
+}
+
+// NewTokenAuth returns a tokenAuth on the given authrorization token.
+func NewTokenAuth(token string) *TokenAuth {
+	return &TokenAuth{token}
+}
+
+func (a *TokenAuth) setAuth(r *http.Request) {
+	if a == nil {
+		return
+	}
+
+	r.Header.Set("Authorization", a.token)
```

Comment:
To be a proper Bearer authentication you must to change the to `fmt.Sprintf("Bearer %s", a.token)`